plugins/node/opentelemetry-instrumentation-nestjs-core/src/instrumentation.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import type * as NestJS from '@nestjs/core';
+import type { NestFactory } from '@nestjs/core/nest-factory.js';
+import type { RouterExecutionContext } from '@nestjs/core/router/router-execution-context.js';
+import type { Controller } from '@nestjs/common/interfaces';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import { VERSION } from './version';
+import { AttributeNames, NestType } from './enums';
+
+export class Instrumentation extends InstrumentationBase<typeof NestJS> {
+  static readonly COMPONENT = '@nestjs/core';
+  static readonly COMMON_ATTRIBUTES = {
+    component: Instrumentation.COMPONENT,
```

Comment:
We are not supposed to be doing the `component` attribute anymore I think.

Reply:
I'd prefer that (and adding the version number of the instrumented lib, which I currently tend to implement again and again) to be done somewhere centrally as well.

Do you have a reference to a discussion somewhere?